### ISSUE TITLE
RPC classes should implement equals() and hashcode() methods

### ISSUE THREAD

**Issue by mikeburke106** [smartdevicelink/sdl_java_suite#116]
In order to compare RPC messages through the `equals()` method, each RPC has to override the `equals()` and `hashcode()` methods.  We need this ability for unit test and for general usability of the RPC classes.  

Eclipse can generate these methods from underlying class variables, but since the current RPC implementation doesn't store anything in class variables, it's all but impossible to implement this in the current design.  Once the feature/refactor_rpcs code is merged into develop, this effort can happen.

**Comment by JulianKast** [comment]
Fixed with #1586
